### PR TITLE
Improve polling client for embedded journal

### PR DESCRIPTION
**Pull request by gpang**

### CODE REVIEW

File: core/common/src/main/java/alluxio/master/PollingMasterInquireClient.java [hunk 1]
```diff
@@ -130,7 +130,8 @@ private InetSocketAddress getAddress() {
         LOG.debug("Timeout while connecting to {}", address);
         continue;
       } catch (AlluxioStatusException e) {
-        throw new RuntimeException(e);
+        LOG.debug("Exception while connecting to {}. error: {}", address, e.getMessage());
```
Comment (bf8086) [inline core/common/src/main/java/alluxio/master/PollingMasterInquireClient.java:133]: Is there a way to aggregate the error here and log/throw it if no valid master is selected?

Comment (gpang) [reply core/common/src/main/java/alluxio/master/PollingMasterInquireClient.java:133]: that already happens elsewhere: https://github.com/Alluxio/alluxio/pull/12376/files#diff-7f60cdbf4f4b4313bee9dd5a28c81ce3577fe6e120c8f8eab522386d7085afe0R112

**Review by bf8086 — APPROVED** [review]
LGTM.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by gpang** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**